kong/timing/hooks/socket.lua

```diff
@@ -0,0 +1,142 @@
+local _M = {}
+
+local old_tcp_connect
+local old_tcp_sslhandshake
+local old_udp_setpeername
+
+local timing
+
+
+local function before_connect(self, host, port, options)
+  local destination
+
+  if string.sub(host, 1, 5) == "unix:" then
+    destination = host
+
+  else
+    destination = "tcp://" .. host .. ":" .. tostring(port)
+  end
+
+  self.__kong_timing_destination__ = destination
+
+  timing.enter_context("connections")
+  timing.enter_context(destination)
+  timing.enter_context("connect")
+end
+
+
+local function after_connect()
+  timing.leave_context() -- leave connect
+  timing.leave_context() -- leave destination
+  timing.leave_context() -- leave connections
+end
+
+
+local function before_sslhandshake(self, reused_session, server_name, _ssl_verify, _send_status_req)
+  timing.enter_context("connections")
+  timing.enter_context(self.__kong_timing_destination__ or "unknown")
+  timing.enter_context("sslhandshake")
+  timing.set_context_prop("attempt_reuse_session", reused_session ~= nil)
+  timing.set_context_prop("sni", server_name)
+end
+
+
+local function after_sslhandshake()
+  timing.leave_context() -- leave sslhandshake
+  timing.leave_context() -- leave destination
+  timing.leave_context() -- leave connections
+end
+
+
+local function before_setpeername(self, host, port)
+  local destination
+
+  if string.sub(host, 1, 5) == "unix:" then
+    destination = host
+
+  else
+    destination = "udp://" .. host .. ":" .. tostring(port)
```

Comment:
tostring(port) -> port, we can concentrate string and number

Reply:
done